documentation/versioning.md

```diff
@@ -0,0 +1,141 @@
+[<< Helpers][1]
+[Traits >>][2]
+
+# Versioning
+
+## Route Macros
+The macros provide a convenient way to manage API versioning in Laravel routes.
+
+They allow you to specify which versions of the API a route is accessible for.
+
+By using these macros, routes can be easily grouped and protected according to version-specific logic,
+ensuring backward compatibility and smooth evolution of the API.
+
+### versionRange
+The versionRange macro allows you to define a range of API versions for which a route is accessible.
+```
+Route::versionRange(VersionEnum::v1, VersionEnum::v2)
+```
+### versionFrom
+This macro helps define the minimum API version required to access a route.
+```
+Route::versionFrom(VersionEnum::v1)
+```
+### versionTo
+The versionTo macro allows you to define the maximum API version for which a route is accessible.
+```
+Route::versionTo(VersionEnum::v2)
+```
+### version
+The version macro allows you to define a specific API version for which a route is accessible.
+```
+Route::version(VersionEnum::v1)
+```
+
+## Middlewares
+
+### CheckVersionMiddleware
+This middleware verifies the requested API version.
+If the version is blocked in the configuration, the application returns HTTP_UPGRADE_STATUS.
+
+To manage blocked versions, configure the `disabled_api_versions` parameter in your application `/app/config/app.php`.
+
+Example:
+```
+'disabled_api_versions' => [
+    VersionEnum::v1->value,
+    VersionEnum::v3->value,
+] 
+```
+### ClearVersion
```

Comment:
let's create another one middleware `VersioningMiddleware` which will call both middlewares and described only it in the readme